src/NServiceBus.AcceptanceTests/Pipeline/When_message_processing_fails.cs

```diff
@@ -0,0 +1,64 @@
+﻿namespace NServiceBus.AcceptanceTests.Pipeline
+{
+    using System;
+    using System.Threading.Tasks;
+    using AcceptanceTesting;
+    using EndpointTemplates;
+    using NUnit.Framework;
+
+    public class When_message_processing_fails : NServiceBusAcceptanceTest
+    {
+        [Test]
+        public async Task Should_provide_additional_exception_information()
+        {
+            var context = await Scenario.Define<Context>()
+                .WithEndpoint<EndpointWithFailingHandler>(e => e
+                    .When(s => s.SendLocal(new FailingMessage()))
+                    .DoNotFailOnErrorMessages())
+                .Done(c => c.MessageFailed)
+                .Run(TimeSpan.FromSeconds(10));
+
+            Assert.IsTrue(context.Exception.Data.Contains("Message type"));
+            Assert.IsTrue(context.Exception.Data.Contains("Handler type"));
+            Assert.IsTrue(context.Exception.Data.Contains("Handler start time"));
+            Assert.IsTrue(context.Exception.Data.Contains("Handler failure time"));
+            Assert.IsTrue(context.Exception.Data.Contains("Message ID"));
```

Comment:
While "Transport message ID" is optional, it would be good to exercise this in a test as well.

Reply:
that's really hard as we don't have anything in the testing framework to check whether the transport uses a dedicated native transport ID